AsyncDisplayKit/ASControlNode.mm

```diff
@@ -280,30 +277,46 @@ - (void)addTarget:(id)target action:(SEL)action forControlEvents:(ASControlNodeE
     {
       // Do we already have an event table for this control event?
       id<NSCopying> eventKey = _ASControlNodeEventKeyForControlEvent(controlEvent);
-      NSMapTable *eventDispatchTable = _controlEventDispatchTable[eventKey];
+      NSMutableOrderedSet *eventTargetActionArray = _controlEventDispatchTable[eventKey];
+      
+      // Create new target action pair
+      ASControlTargetAction *targetAtion = [[ASControlTargetAction alloc] init];
+      targetAtion.action = action;
+      targetAtion.target = target;
+      
       // Create it if necessary.
-      if (!eventDispatchTable)
-      {
-        // Create the dispatch table for this event.
-        eventDispatchTable = [NSMapTable weakToStrongObjectsMapTable];
-        if (eventKey) {
-          [_controlEventDispatchTable setObject:eventDispatchTable forKey:eventKey];
+      if (eventTargetActionArray) {
+        // UIControl does not support duplicate target-action-events entries, so we replicate that behavior.
+        // See: https://github.com/facebook/AsyncDisplayKit/files/205466/DuplicateActionsTest.playground.zip
+        NSMutableOrderedSet *newEventTargetActionArray;
+        
+        for (ASControlTargetAction *storedTargetAction in eventTargetActionArray) {
+          if ([storedTargetAction isEqual:targetAtion]) {
+            continue;
+          }
+          
+          if (!newEventTargetActionArray) {
+            newEventTargetActionArray = [[NSMutableOrderedSet alloc] init];
+          }
+          
+          [newEventTargetActionArray addObject:storedTargetAction];
+        }
+        
+        if (!newEventTargetActionArray) {
+          newEventTargetActionArray = [[NSMutableOrderedSet alloc] init];
         }
+        
+        eventTargetActionArray = newEventTargetActionArray;
```

Comment:
@Adlai-Holler Yes, I totally forgot that now we're using `NSMutableOrderedSet`, so removing and adding an item will do that same.